src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -256,22 +263,19 @@ class ContentOpenWith extends PureComponent<Props, State> {
                 this.setState({ executePostData: executeData });
                 break;
             case HTTP_GET:
-                // window.open() is intentionally invoked with no URL to support workaround below
-                const windowRef = window.open();
-                if (!windowRef) {
+                if (!this.windowRef) {
                     this.executeIntegrationErrorHandler(
                         Error(WINDOW_OPEN_BLOCKED_ERROR),
                     );
                     return;
-                } else {
-                    windowRef.location = url;
-                    windowRef.opener = null;
-                    this.onExecute();
                 }
+
                 // Prevents abuse of window.opener
                 // see here for more details: https://mathiasbynens.github.io/rel-noopener/
-                windowRef.opener = null;
-                windowRef.location = url;
+                this.windowRef.location = url;
```

Comment:
Could also be part of your windowUtils.js

setLocation(window, string?)
setOpener(window, string?)

Reply:
It seems like mocking the global.window property would be more efficient. Otherwise you're basically writing a wrapper around every window property.

Reply:
Mocking global properties can cause tests to fail unless the original version is properly restored

Reply:
Take care to reset it, if it's even necessary. It seems like it'd be worth a little experimentation. Also, it looks like `global` _is_ `window` in Jest. Maybe it already mocks these properties?

https://github.com/facebook/jest/issues/3692#issuecomment-408132529

Reply:
As an alternative, one strategy I’ve used in the past is to store a reference as `this.window = window` via `componentDidMount`.

Reply:
@priyajeet do you see value in a separate constants file outside of testability? `this.window = window` would solve that imo.

Reply:
The window/document utility can be limited to just this, and not needed to add wrapper for every property set

```js
// @flow

function getWindow() {
    return window;
}

function getDocument() {
    return document;
}

export { getWindow, getDocument };
```

Reply:
saved this.window instance also sounds good too.
Though I would just do this.window = getWindow() in that case if the componentWillMount() function is tested, in which case you will be like expect(getWindow).to.be.called()

Reply:
is the window object not available in jest? 

```componentDidMount();
expect(this.window).toEqual(window);```

Reply:
I think its `global` instead